public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -246,13 +246,11 @@ function ExpendedRow({ correlation: { source, target, ...correlation }, readOnly
     []
   );
 
-  return (
-    <EditCorrelationForm
-      correlation={{ ...correlation, sourceUID: source.uid, targetUID: target?.uid }}
-      onUpdated={onUpdated}
-      readOnly={readOnly}
-    />
-  );
+  let corr: Correlation = target?.uid
```

Comment:
We need to be a bit more specific about what kind of correlation we are creating which means a bit more typing but hopefully also more type safety.